www/js/diary/LabelTab.tsx

```diff
@@ -87,17 +86,25 @@ const LabelTab = () => {
     }
   }, [appConfig, refreshTime]);
 
-  // whenever timelineMap is updated, update the displayedEntries
-  // according to the active filter
+  // whenever timelineMap is updated, map unprocessed inputs to timeline entries, and
+  // update the displayedEntries according to the active filter
```

Comment:
so this will be called every time the data is reloaded, and also that when we get new user inputs. But it looks like it maps *all* inputs to timeline entries. Is there a way to only do so for new inputs - i.e. incrementally?

Reply:
Theoretically, I think so. But we'd have to be really careful about it because input matching relies on `nextEntry` in some cases.

If you load only week 1, the last trip of week 1 has no `nextEntry`. Once you load week 2, that last trip of week 1 still needs to be re-matched because the input matched to it before might actually belong on the first trip of week 2. If we didn't re-match the last trip of week 1, we could end up with a duplicate or something.

So I think the *safest* thing to do is just treat input matching holistically.

And I suspect network/ retrieval time is by far the bottleneck for Label tab slowness - any performance savings here are negligible compared to that